packages/forms/src/UIForm/merge.js

```diff
@@ -0,0 +1,207 @@
+			};
+		}
+		if (get(props, 'schema.schema.enum')) {
+			newProps.schema.enum = props.schema.schema.enum;
+		}
+		const onChange = props.onChange;
+		newProps.onChange = (event, payload) => {
+			if (!payload) {
+				onChange({}, { schema: newProps.schema, value: event });
+			} else {
+				onChange(event, payload);
+			}
+		};
+		newProps.required = props.schema.required;
+		newProps.placeholder = props.schema.placeholder;
+		newProps.options = props.schema.options;
+		return (
+			<FieldTemplate label={props.schema.title} id={newProps.id}>
+				<Component {...newProps} />
+			</FieldTemplate>
+		);
+	}
+	TFMigratedWidget.propTypes = {
+		formContext: PropTypes.object,
+		onChange: PropTypes.func,
+		schema: PropTypes.object,
+		titleMap: PropTypes.array,
+	};
+	TFMigratedWidget.displayName = 'TFMigratedWidget';
+	return TFMigratedWidget;
+};
+
+function updateWidgets(items, uiSchema, widgets, prefix) {
+	return items.map(ui => {
+		const uiSchemaKey = ui.key.replace(prefix, '');
+		if (ui.key && uiSchema[uiSchemaKey]) {
+			const config = uiSchema[uiSchemaKey];
+			Object.keys(config || {}).forEach(configKey => {
+				if (!configKey.startsWith('ui:')) {
+					return;
+				}
+				if (configKey === 'ui:widget') {
+					const uiWidget = config[configKey];
+					if (typeof uiWidget !== 'string') {
+						widgets[uiWidget.name || ui.key] = wrapCustomWidget(uiWidget);
+						ui.widget = uiWidget.name || ui.key;
+					} else {
+						ui.widget = uiWidget;
+						if (uiWidget === 'updown') {
+							ui.widget = 'text';
+							ui.type = 'number';
+						} else if (uiWidget === 'password') {
+							ui.widget = 'text';
+							ui.type = 'password';
+						} else if (uiWidget === 'url') {
+							ui.widget = 'text';
+							ui.type = 'url';
+						} else if (uiWidget === 'color') {
+							ui.widget = 'text';
+							ui.type = 'color';
+						} else if (uiWidget === 'radio') {
+							ui.widget = 'radios';
+						} else if (uiWidget === 'enumeration') {
+							delete ui.items;
+						}
+					}
+				} else if (configKey === 'ui:placeholder') {
+					ui.placeholder = config['ui:placeholder'];
+				} else if (configKey === 'ui:options') {
+					ui.options = config['ui:options'];
+				} else if (configKey === 'ui:autofocus') {
+					ui.autoFocus = config['ui:autofocus'];
+				} else if (configKey === 'ui:help') {
+					ui.helpvalue = config['ui:help'];
+				} else if (config['ui:trigger']) {
+					ui.triggers = config['ui:trigger'];
+				} else {
+					ui[configKey.replace('ui:', '')] = config[configKey];
+				}
+			});
+			if (ui.items && config && ui.widget !== 'enumeration') {
+				ui.items = updateWidgets(ui.items, config, widgets, `${ui.key}.`);
+			}
+		}
+		return ui;
+	});
+}
+
+/**
+ * migrate from react-jsonschema-form to UIschema
+ * @param {Object} mergedSchema
+ */
+export function migrate(jsonSchema, uiSchema) {
+	const safeUISchema = parseProperties(jsonSchema, true);
+	const widgets = {};
+	if (!uiSchema || uiSchema === {}) {
```

Comment:
`uiSchema === {}` is always false
BTW this test should be on top of the function. Calling parseProperties is useless, and should return uiSchema as an empty array.

Reply:
uiSchema === {} is useless true but
we need a uiSchema in all case to handle enumNames which is not part of the spec